architecture.md

```diff
@@ -0,0 +1,31 @@
+# os-autoinst architecture
+This document gives an overview about the multi-process architectue of os-autoinst.
+
+## Process tree
+Once everything is running, the process tree looks like this:
+
+* **isotovideo**: spawns further processes, IO-loop for passing commands (main occupation), cleanup  
+  relevant files: `isotovideo`
+
+    * **backend**: spawns and handles backend (eg. qemu), receives commands from isotovideo IO-loop  
```

Comment:
I would mention (as it's a quite important task) that it handles the VNC connections and makes regular screenshots.